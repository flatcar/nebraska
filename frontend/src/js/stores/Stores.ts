import ActivityStore from './ActivityStore';
import ApplicationsStore from './ApplicationsStore';
import GroupChartsStore from './GroupChartsStore';

export const applicationsStore = new ApplicationsStore;
export const activityStore = new ActivityStore;
export const groupChartStore = new GroupChartsStore();
