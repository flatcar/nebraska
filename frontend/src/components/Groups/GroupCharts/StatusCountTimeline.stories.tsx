import { Meta, StoryFn } from '@storybook/react';
import { JSX } from 'react/jsx-runtime';
import { MemoryRouter } from 'react-router-dom';

import GroupChartsStore from '../../../stores/GroupChartsStore';
import { groupChartStoreContext } from '../../../stores/Stores';
import StatusCountTimeline, { StatusCountTimelineProps } from './StatusCountTimeline';

export default {
  title: 'groups/StatusCountTimeline',
} as Meta;

const statusTimelineData = {
  '2021-11-07T10:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T11:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T12:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T13:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T14:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T15:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T16:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T17:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T18:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T19:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T20:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T21:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T22:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-07T23:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-08T00:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-08T01:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-08T02:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-08T03:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-08T04:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-08T05:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-08T06:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-08T07:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-08T08:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
  '2021-11-08T09:35:28.823161+01:00': {
    '1': { '2191.5.0': 177 },
    '2': { '2191.5.0': 221 },
    '3': { '2191.5.0': 12 },
    '6': { '2191.5.0': 193 },
    '7': { '2191.5.0': 215 },
  },
  '2021-11-08T10:35:28.823161+01:00': { '1': {}, '2': {}, '3': {}, '6': {}, '7': {} },
};

const Template: StoryFn<StatusCountTimelineProps> = (
  args: JSX.IntrinsicAttributes & StatusCountTimelineProps
) => {
  class GroupChartsStoreMock extends GroupChartsStore {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    async getGroupStatusCountTimeline(_appID: string, _groupID: string, _duration: string) {
      return statusTimelineData;
    }
  }

  const ChartStoreContext = groupChartStoreContext();

  return (
    <ChartStoreContext.Provider value={new GroupChartsStoreMock()}>
      <MemoryRouter>
        <StatusCountTimeline {...args} />
      </MemoryRouter>
    </ChartStoreContext.Provider>
  );
};

export const Timeline = {
  render: Template,

  args: {
    isAnimationActive: false,
    group: {
      id: '9a2deb70-37be-4026-853f-bfdd6b347bbe',
      name: 'Stable (AMD64)',
      description: 'For production clusters (AMD64)',
      created_ts: '2015-09-19T07:09:34.269062+02:00',
      rollout_in_progress: true,
      application_id: 'e96281a6-d1af-4bde-9a0a-97b76e56dc57',
      channel_id: 'e06064ad-4414-4904-9a6e-fd465593d1b2',
      policy_updates_enabled: true,
      policy_safe_mode: false,
      policy_office_hours: false,
      policy_timezone: 'Europe/Berlin',
      policy_period_interval: '1 minutes',
      policy_max_updates_per_period: 999999,
      policy_update_timeout: '60 minutes',
      channel: {
        id: 'e06064ad-4414-4904-9a6e-fd465593d1b2',
        name: 'stable',
        color: '#14b9d6',
        created_ts: '2015-09-19T07:09:34.261241+02:00',
        application_id: 'e96281a6-d1af-4bde-9a0a-97b76e56dc57',
        package_id: '84b4c599-9b6b-44a8-b13c-d4263fff0403',
        package: {
          id: '84b4c599-9b6b-44a8-b13c-d4263fff0403',
          type: 1,
          version: '2191.5.0',
          url: 'https://update.release.flatcar-linux.net/amd64-usr/2191.5.0/',
          filename: 'flatcar_production_update.gz',
          description: 'Flatcar Container Linux 2191.5.0',
          size: '465881871',
          hash: 'r3nufcxgMTZaxYEqL+x2zIoeClk=',
          created_ts: '2019-09-05T12:41:09.265687+02:00',
          channels_blacklist: null,
          application_id: 'e96281a6-d1af-4bde-9a0a-97b76e56dc57',
          flatcar_action: {
            id: '1f6e1bcf-4ebb-4fe6-8ca3-2cb6ad90d5dd',
            event: 'postinstall',
            chromeos_version: '',
            sha256: 'LIkAKVZY2EJFiwTmltiJZLFLA5xT/FodbjVgqkyF/y8=',
            needs_admin: false,
            is_delta: false,
            disable_payload_backoff: true,
            metadata_signature_rsa: '',
            metadata_size: '',
            deadline: '',
            created_ts: '2019-08-20T02:12:37.532281+02:00',
          },
          arch: 1,
          extra_files: [],
        },
        arch: 1,
      },
      track: 'stable',
    },
    duration: { displayValue: '1 day', queryValue: '1d', disabled: false },
  },
};
