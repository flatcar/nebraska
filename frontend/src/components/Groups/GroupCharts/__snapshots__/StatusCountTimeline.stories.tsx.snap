// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`groups/StatusCountTimeline Timeline smoke-test 1`] = `
<div class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-mwb7rv-MuiGrid-root">
  <div class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root">
    <div class="recharts-wrapper"
         style="position: relative; cursor: default; width: 500px; height: 400px;"
    >
      <svg class="recharts-surface"
           width="500"
           height="400"
           viewbox="0 0 500 400"
           style="width: 100%; height: 100%;"
      >
        <title>
        </title>
        <desc>
        </desc>
        <defs>
          <clipPath id="recharts1-clip">
            <rect x="60"
                  y="10"
                  height="360"
                  width="410"
            >
            </rect>
          </clipPath>
        </defs>
        <g class="recharts-cartesian-grid">
          <g class="recharts-cartesian-grid-horizontal">
            <line stroke-dasharray="3 3"
                  stroke="#ccc"
                  fill="none"
                  x="60"
                  y="10"
                  width="410"
                  height="360"
                  x1="60"
                  y1="370"
                  x2="470"
                  y2="370"
            >
            </line>
            <line stroke-dasharray="3 3"
                  stroke="#ccc"
                  fill="none"
                  x="60"
                  y="10"
                  width="410"
                  height="360"
                  x1="60"
                  y1="280"
                  x2="470"
                  y2="280"
            >
            </line>
            <line stroke-dasharray="3 3"
                  stroke="#ccc"
                  fill="none"
                  x="60"
                  y="10"
                  width="410"
                  height="360"
                  x1="60"
                  y1="190"
                  x2="470"
                  y2="190"
            >
            </line>
            <line stroke-dasharray="3 3"
                  stroke="#ccc"
                  fill="none"
                  x="60"
                  y="10"
                  width="410"
                  height="360"
                  x1="60"
                  y1="100"
                  x2="470"
                  y2="100"
            >
            </line>
            <line stroke-dasharray="3 3"
                  stroke="#ccc"
                  fill="none"
                  x="60"
                  y="10"
                  width="410"
                  height="360"
                  x1="60"
                  y1="10"
                  x2="470"
                  y2="10"
            >
            </line>
          </g>
          <g class="recharts-cartesian-grid-vertical">
            <line stroke-dasharray="3 3"
                  stroke="#ccc"
                  fill="none"
                  x="60"
                  y="10"
                  width="410"
                  height="360"
                  x1="203.64794641203702"
                  y1="10"
                  x2="203.64794641203702"
                  y2="370"
            >
            </line>
            <line stroke-dasharray="3 3"
                  stroke="#ccc"
                  fill="none"
                  x="60"
                  y="10"
                  width="410"
                  height="360"
                  x1="101.14794641203703"
                  y1="10"
                  x2="101.14794641203703"
                  y2="370"
            >
            </line>
            <line stroke-dasharray="3 3"
                  stroke="#ccc"
                  fill="none"
                  x="60"
                  y="10"
                  width="410"
                  height="360"
                  x1="408.6479464120371"
                  y1="10"
                  x2="408.6479464120371"
                  y2="370"
            >
            </line>
            <line stroke-dasharray="3 3"
                  stroke="#ccc"
                  fill="none"
                  x="60"
                  y="10"
                  width="410"
                  height="360"
                  x1="306.1479464120371"
                  y1="10"
                  x2="306.1479464120371"
                  y2="370"
            >
            </line>
            <line stroke-dasharray="3 3"
                  stroke="#ccc"
                  fill="none"
                  x="60"
                  y="10"
                  width="410"
                  height="360"
                  x1="60"
                  y1="10"
                  x2="60"
                  y2="370"
            >
            </line>
            <line stroke-dasharray="3 3"
                  stroke="#ccc"
                  fill="none"
                  x="60"
                  y="10"
                  width="410"
                  height="360"
                  x1="470"
                  y1="10"
                  x2="470"
                  y2="370"
            >
            </line>
          </g>
        </g>
        <g class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis">
          <line orientation="bottom"
                width="410"
                height="30"
                stroke="#000"
                x="60"
                y="370"
                class="recharts-cartesian-axis-line"
                fill="none"
                x1="60"
                y1="370"
                x2="470"
                y2="370"
          >
          </line>
          <g class="recharts-cartesian-axis-ticks">
            <g class="recharts-layer recharts-cartesian-axis-tick">
              <line orientation="bottom"
                    width="410"
                    height="30"
                    stroke="#000"
                    x="60"
                    y="370"
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    x1="203.64794641203702"
                    y1="376"
                    x2="203.64794641203702"
                    y2="370"
              >
              </line>
              <text orientation="bottom"
                    width="410"
                    height="30"
                    stroke="none"
                    x="203.64794641203702"
                    y="378"
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    text-anchor="middle"
                    fill="#000"
              >
                <tspan x="203.64794641203702"
                       dy="0.71em"
                >
                  06:00 PM
                </tspan>
              </text>
            </g>
            <g class="recharts-layer recharts-cartesian-axis-tick">
              <line orientation="bottom"
                    width="410"
                    height="30"
                    stroke="#000"
                    x="60"
                    y="370"
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    x1="101.14794641203703"
                    y1="376"
                    x2="101.14794641203703"
                    y2="370"
              >
              </line>
              <text orientation="bottom"
                    width="410"
                    height="30"
                    stroke="none"
                    x="101.14794641203703"
                    y="378"
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    text-anchor="middle"
                    fill="#000"
              >
                <tspan x="101.14794641203703"
                       dy="0.71em"
                >
                  12:00 PM
                </tspan>
              </text>
            </g>
            <g class="recharts-layer recharts-cartesian-axis-tick">
              <line orientation="bottom"
                    width="410"
                    height="30"
                    stroke="#000"
                    x="60"
                    y="370"
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    x1="408.6479464120371"
                    y1="376"
                    x2="408.6479464120371"
                    y2="370"
              >
              </line>
              <text orientation="bottom"
                    width="410"
                    height="30"
                    stroke="none"
                    x="408.6479464120371"
                    y="378"
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    text-anchor="middle"
                    fill="#000"
              >
                <tspan x="408.6479464120371"
                       dy="0.71em"
                >
                  06:00 AM
                </tspan>
              </text>
            </g>
            <g class="recharts-layer recharts-cartesian-axis-tick">
              <line orientation="bottom"
                    width="410"
                    height="30"
                    stroke="#000"
                    x="60"
                    y="370"
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    x1="306.1479464120371"
                    y1="376"
                    x2="306.1479464120371"
                    y2="370"
              >
              </line>
              <text orientation="bottom"
                    width="410"
                    height="30"
                    stroke="none"
                    x="306.1479464120371"
                    y="378"
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    text-anchor="middle"
                    fill="#000"
              >
                <tspan x="306.1479464120371"
                       dy="0.71em"
                >
                  Nov 8
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis">
          <line orientation="left"
                width="60"
                height="360"
                stroke="#000"
                x="0"
                y="10"
                class="recharts-cartesian-axis-line"
                fill="none"
                x1="60"
                y1="10"
                x2="60"
                y2="370"
          >
          </line>
          <g class="recharts-cartesian-axis-ticks">
            <g class="recharts-layer recharts-cartesian-axis-tick">
              <line orientation="left"
                    width="60"
                    height="360"
                    stroke="#000"
                    x="0"
                    y="10"
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    x1="54"
                    y1="370"
                    x2="60"
                    y2="370"
              >
              </line>
              <text orientation="left"
                    width="60"
                    height="360"
                    stroke="none"
                    x="52"
                    y="370"
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    text-anchor="end"
                    fill="#000"
              >
                <tspan x="52"
                       dy="0.355em"
                >
                  0
                </tspan>
              </text>
            </g>
            <g class="recharts-layer recharts-cartesian-axis-tick">
              <line orientation="left"
                    width="60"
                    height="360"
                    stroke="#000"
                    x="0"
                    y="10"
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    x1="54"
                    y1="280"
                    x2="60"
                    y2="280"
              >
              </line>
              <text orientation="left"
                    width="60"
                    height="360"
                    stroke="none"
                    x="52"
                    y="280"
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    text-anchor="end"
                    fill="#000"
              >
                <tspan x="52"
                       dy="0.355em"
                >
                  250
                </tspan>
              </text>
            </g>
            <g class="recharts-layer recharts-cartesian-axis-tick">
              <line orientation="left"
                    width="60"
                    height="360"
                    stroke="#000"
                    x="0"
                    y="10"
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    x1="54"
                    y1="190"
                    x2="60"
                    y2="190"
              >
              </line>
              <text orientation="left"
                    width="60"
                    height="360"
                    stroke="none"
                    x="52"
                    y="190"
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    text-anchor="end"
                    fill="#000"
              >
                <tspan x="52"
                       dy="0.355em"
                >
                  500
                </tspan>
              </text>
            </g>
            <g class="recharts-layer recharts-cartesian-axis-tick">
              <line orientation="left"
                    width="60"
                    height="360"
                    stroke="#000"
                    x="0"
                    y="10"
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    x1="54"
                    y1="100"
                    x2="60"
                    y2="100"
              >
              </line>
              <text orientation="left"
                    width="60"
                    height="360"
                    stroke="none"
                    x="52"
                    y="100"
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    text-anchor="end"
                    fill="#000"
              >
                <tspan x="52"
                       dy="0.355em"
                >
                  750
                </tspan>
              </text>
            </g>
            <g class="recharts-layer recharts-cartesian-axis-tick">
              <line orientation="left"
                    width="60"
                    height="360"
                    stroke="#000"
                    x="0"
                    y="10"
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    x1="54"
                    y1="10"
                    x2="60"
                    y2="10"
              >
              </line>
              <text orientation="left"
                    width="60"
                    height="360"
                    stroke="none"
                    x="52"
                    y="10"
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    text-anchor="end"
                    fill="#000"
              >
                <tspan x="52"
                       dy="0.355em"
                >
                  1000
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g class="recharts-layer recharts-area">
          <g class="recharts-layer">
            <path cursor="pointer"
                  fill="rgb(89, 89, 89)"
                  fill-opacity="0.6"
                  width="410"
                  height="360"
                  stroke="none"
                  class="recharts-curve recharts-area-area"
                  d="M60,370L68.542,370L68.542,370L85.625,370L85.625,370L102.708,370L102.708,370L119.792,370L119.792,370L136.875,370L136.875,370L153.958,370L153.958,370L171.042,370L171.042,370L188.125,370L188.125,370L205.208,370L205.208,370L222.292,370L222.292,370L239.375,370L239.375,370L256.458,370L256.458,370L273.542,370L273.542,370L290.625,370L290.625,370L307.708,370L307.708,370L324.792,370L324.792,370L341.875,370L341.875,370L358.958,370L358.958,370L376.042,370L376.042,370L393.125,370L393.125,370L410.208,370L410.208,370L427.292,370L427.292,370L444.375,370L444.375,306.28L461.458,306.28L461.458,370L470,370L470,370L461.458,370L461.458,370L444.375,370L444.375,370L427.292,370L427.292,370L410.208,370L410.208,370L393.125,370L393.125,370L376.042,370L376.042,370L358.958,370L358.958,370L341.875,370L341.875,370L324.792,370L324.792,370L307.708,370L307.708,370L290.625,370L290.625,370L273.542,370L273.542,370L256.458,370L256.458,370L239.375,370L239.375,370L222.292,370L222.292,370L205.208,370L205.208,370L188.125,370L188.125,370L171.042,370L171.042,370L153.958,370L153.958,370L136.875,370L136.875,370L119.792,370L119.792,370L102.708,370L102.708,370L85.625,370L85.625,370L68.542,370L68.542,370L60,370Z"
            >
            </path>
            <path stroke="rgb(89, 89, 89)"
                  cursor="pointer"
                  fill="none"
                  fill-opacity="0.6"
                  width="410"
                  height="360"
                  class="recharts-curve recharts-area-curve"
                  d="M60,370L68.542,370L68.542,370L85.625,370L85.625,370L102.708,370L102.708,370L119.792,370L119.792,370L136.875,370L136.875,370L153.958,370L153.958,370L171.042,370L171.042,370L188.125,370L188.125,370L205.208,370L205.208,370L222.292,370L222.292,370L239.375,370L239.375,370L256.458,370L256.458,370L273.542,370L273.542,370L290.625,370L290.625,370L307.708,370L307.708,370L324.792,370L324.792,370L341.875,370L341.875,370L358.958,370L358.958,370L376.042,370L376.042,370L393.125,370L393.125,370L410.208,370L410.208,370L427.292,370L427.292,370L444.375,370L444.375,306.28L461.458,306.28L461.458,370L470,370"
            >
            </path>
          </g>
        </g>
        <g class="recharts-layer recharts-area">
          <g class="recharts-layer">
            <path cursor="pointer"
                  fill="#3182bd"
                  fill-opacity="0.6"
                  width="410"
                  height="360"
                  stroke="none"
                  class="recharts-curve recharts-area-area"
                  d="M60,370L68.542,370L68.542,370L85.625,370L85.625,370L102.708,370L102.708,370L119.792,370L119.792,370L136.875,370L136.875,370L153.958,370L153.958,370L171.042,370L171.042,370L188.125,370L188.125,370L205.208,370L205.208,370L222.292,370L222.292,370L239.375,370L239.375,370L256.458,370L256.458,370L273.542,370L273.542,370L290.625,370L290.625,370L307.708,370L307.708,370L324.792,370L324.792,370L341.875,370L341.875,370L358.958,370L358.958,370L376.042,370L376.042,370L393.125,370L393.125,370L410.208,370L410.208,370L427.292,370L427.292,370L444.375,370L444.375,226.72L461.458,226.72L461.458,370L470,370L470,370L461.458,370L461.458,306.28L444.375,306.28L444.375,370L427.292,370L427.292,370L410.208,370L410.208,370L393.125,370L393.125,370L376.042,370L376.042,370L358.958,370L358.958,370L341.875,370L341.875,370L324.792,370L324.792,370L307.708,370L307.708,370L290.625,370L290.625,370L273.542,370L273.542,370L256.458,370L256.458,370L239.375,370L239.375,370L222.292,370L222.292,370L205.208,370L205.208,370L188.125,370L188.125,370L171.042,370L171.042,370L153.958,370L153.958,370L136.875,370L136.875,370L119.792,370L119.792,370L102.708,370L102.708,370L85.625,370L85.625,370L68.542,370L68.542,370L60,370Z"
            >
            </path>
            <path stroke="#3182bd"
                  cursor="pointer"
                  fill="none"
                  fill-opacity="0.6"
                  width="410"
                  height="360"
                  class="recharts-curve recharts-area-curve"
                  d="M60,370L68.542,370L68.542,370L85.625,370L85.625,370L102.708,370L102.708,370L119.792,370L119.792,370L136.875,370L136.875,370L153.958,370L153.958,370L171.042,370L171.042,370L188.125,370L188.125,370L205.208,370L205.208,370L222.292,370L222.292,370L239.375,370L239.375,370L256.458,370L256.458,370L273.542,370L273.542,370L290.625,370L290.625,370L307.708,370L307.708,370L324.792,370L324.792,370L341.875,370L341.875,370L358.958,370L358.958,370L376.042,370L376.042,370L393.125,370L393.125,370L410.208,370L410.208,370L427.292,370L427.292,370L444.375,370L444.375,226.72L461.458,226.72L461.458,370L470,370"
            >
            </path>
          </g>
        </g>
        <g class="recharts-layer recharts-area">
          <g class="recharts-layer">
            <path cursor="pointer"
                  fill="rgba(164,45,36,1)"
                  fill-opacity="0.6"
                  width="410"
                  height="360"
                  stroke="none"
                  class="recharts-curve recharts-area-area"
                  d="M60,370L68.542,370L68.542,370L85.625,370L85.625,370L102.708,370L102.708,370L119.792,370L119.792,370L136.875,370L136.875,370L153.958,370L153.958,370L171.042,370L171.042,370L188.125,370L188.125,370L205.208,370L205.208,370L222.292,370L222.292,370L239.375,370L239.375,370L256.458,370L256.458,370L273.542,370L273.542,370L290.625,370L290.625,370L307.708,370L307.708,370L324.792,370L324.792,370L341.875,370L341.875,370L358.958,370L358.958,370L376.042,370L376.042,370L393.125,370L393.125,370L410.208,370L410.208,370L427.292,370L427.292,370L444.375,370L444.375,222.4L461.458,222.4L461.458,370L470,370L470,370L461.458,370L461.458,226.72L444.375,226.72L444.375,370L427.292,370L427.292,370L410.208,370L410.208,370L393.125,370L393.125,370L376.042,370L376.042,370L358.958,370L358.958,370L341.875,370L341.875,370L324.792,370L324.792,370L307.708,370L307.708,370L290.625,370L290.625,370L273.542,370L273.542,370L256.458,370L256.458,370L239.375,370L239.375,370L222.292,370L222.292,370L205.208,370L205.208,370L188.125,370L188.125,370L171.042,370L171.042,370L153.958,370L153.958,370L136.875,370L136.875,370L119.792,370L119.792,370L102.708,370L102.708,370L85.625,370L85.625,370L68.542,370L68.542,370L60,370Z"
            >
            </path>
            <path stroke="rgba(164,45,36,1)"
                  cursor="pointer"
                  fill="none"
                  fill-opacity="0.6"
                  width="410"
                  height="360"
                  class="recharts-curve recharts-area-curve"
                  d="M60,370L68.542,370L68.542,370L85.625,370L85.625,370L102.708,370L102.708,370L119.792,370L119.792,370L136.875,370L136.875,370L153.958,370L153.958,370L171.042,370L171.042,370L188.125,370L188.125,370L205.208,370L205.208,370L222.292,370L222.292,370L239.375,370L239.375,370L256.458,370L256.458,370L273.542,370L273.542,370L290.625,370L290.625,370L307.708,370L307.708,370L324.792,370L324.792,370L341.875,370L341.875,370L358.958,370L358.958,370L376.042,370L376.042,370L393.125,370L393.125,370L410.208,370L410.208,370L427.292,370L427.292,370L444.375,370L444.375,222.4L461.458,222.4L461.458,370L470,370"
            >
            </path>
          </g>
        </g>
        <g class="recharts-layer recharts-area">
          <g class="recharts-layer">
            <path cursor="pointer"
                  fill="rgba(40,95,43,1)"
                  fill-opacity="0.6"
                  width="410"
                  height="360"
                  stroke="none"
                  class="recharts-curve recharts-area-area"
                  d="M60,370L68.542,370L68.542,370L85.625,370L85.625,370L102.708,370L102.708,370L119.792,370L119.792,370L136.875,370L136.875,370L153.958,370L153.958,370L171.042,370L171.042,370L188.125,370L188.125,370L205.208,370L205.208,370L222.292,370L222.292,370L239.375,370L239.375,370L256.458,370L256.458,370L273.542,370L273.542,370L290.625,370L290.625,370L307.708,370L307.708,370L324.792,370L324.792,370L341.875,370L341.875,370L358.958,370L358.958,370L376.042,370L376.042,370L393.125,370L393.125,370L410.208,370L410.208,370L427.292,370L427.292,370L444.375,370L444.375,152.92L461.458,152.92L461.458,370L470,370L470,370L461.458,370L461.458,222.4L444.375,222.4L444.375,370L427.292,370L427.292,370L410.208,370L410.208,370L393.125,370L393.125,370L376.042,370L376.042,370L358.958,370L358.958,370L341.875,370L341.875,370L324.792,370L324.792,370L307.708,370L307.708,370L290.625,370L290.625,370L273.542,370L273.542,370L256.458,370L256.458,370L239.375,370L239.375,370L222.292,370L222.292,370L205.208,370L205.208,370L188.125,370L188.125,370L171.042,370L171.042,370L153.958,370L153.958,370L136.875,370L136.875,370L119.792,370L119.792,370L102.708,370L102.708,370L85.625,370L85.625,370L68.542,370L68.542,370L60,370Z"
            >
            </path>
            <path stroke="rgba(40,95,43,1)"
                  cursor="pointer"
                  fill="none"
                  fill-opacity="0.6"
                  width="410"
                  height="360"
                  class="recharts-curve recharts-area-curve"
                  d="M60,370L68.542,370L68.542,370L85.625,370L85.625,370L102.708,370L102.708,370L119.792,370L119.792,370L136.875,370L136.875,370L153.958,370L153.958,370L171.042,370L171.042,370L188.125,370L188.125,370L205.208,370L205.208,370L222.292,370L222.292,370L239.375,370L239.375,370L256.458,370L256.458,370L273.542,370L273.542,370L290.625,370L290.625,370L307.708,370L307.708,370L324.792,370L324.792,370L341.875,370L341.875,370L358.958,370L358.958,370L376.042,370L376.042,370L393.125,370L393.125,370L410.208,370L410.208,370L427.292,370L427.292,370L444.375,370L444.375,152.92L461.458,152.92L461.458,370L470,370"
            >
            </path>
          </g>
        </g>
        <g class="recharts-layer recharts-area">
          <g class="recharts-layer">
            <path cursor="pointer"
                  fill="rgba(17,40,141,1)"
                  fill-opacity="0.6"
                  width="410"
                  height="360"
                  stroke="none"
                  class="recharts-curve recharts-area-area"
                  d="M60,370L68.542,370L68.542,370L85.625,370L85.625,370L102.708,370L102.708,370L119.792,370L119.792,370L136.875,370L136.875,370L153.958,370L153.958,370L171.042,370L171.042,370L188.125,370L188.125,370L205.208,370L205.208,370L222.292,370L222.292,370L239.375,370L239.375,370L256.458,370L256.458,370L273.542,370L273.542,370L290.625,370L290.625,370L307.708,370L307.708,370L324.792,370L324.792,370L341.875,370L341.875,370L358.958,370L358.958,370L376.042,370L376.042,370L393.125,370L393.125,370L410.208,370L410.208,370L427.292,370L427.292,370L444.375,370L444.375,75.52L461.458,75.52L461.458,370L470,370L470,370L461.458,370L461.458,152.92L444.375,152.92L444.375,370L427.292,370L427.292,370L410.208,370L410.208,370L393.125,370L393.125,370L376.042,370L376.042,370L358.958,370L358.958,370L341.875,370L341.875,370L324.792,370L324.792,370L307.708,370L307.708,370L290.625,370L290.625,370L273.542,370L273.542,370L256.458,370L256.458,370L239.375,370L239.375,370L222.292,370L222.292,370L205.208,370L205.208,370L188.125,370L188.125,370L171.042,370L171.042,370L153.958,370L153.958,370L136.875,370L136.875,370L119.792,370L119.792,370L102.708,370L102.708,370L85.625,370L85.625,370L68.542,370L68.542,370L60,370Z"
            >
            </path>
            <path stroke="rgba(17,40,141,1)"
                  cursor="pointer"
                  fill="none"
                  fill-opacity="0.6"
                  width="410"
                  height="360"
                  class="recharts-curve recharts-area-curve"
                  d="M60,370L68.542,370L68.542,370L85.625,370L85.625,370L102.708,370L102.708,370L119.792,370L119.792,370L136.875,370L136.875,370L153.958,370L153.958,370L171.042,370L171.042,370L188.125,370L188.125,370L205.208,370L205.208,370L222.292,370L222.292,370L239.375,370L239.375,370L256.458,370L256.458,370L273.542,370L273.542,370L290.625,370L290.625,370L307.708,370L307.708,370L324.792,370L324.792,370L341.875,370L341.875,370L358.958,370L358.958,370L376.042,370L376.042,370L393.125,370L393.125,370L410.208,370L410.208,370L427.292,370L427.292,370L444.375,370L444.375,75.52L461.458,75.52L461.458,370L470,370"
            >
            </path>
          </g>
        </g>
      </svg>
      <div tabindex="-1"
           class="recharts-tooltip-wrapper recharts-tooltip-wrapper-right recharts-tooltip-wrapper-bottom"
           style="visibility: hidden; pointer-events: none; position: absolute; top: 0px; left: 0px; transform: translate(60px, 10px);"
      >
        <div class="custom-tooltip">
          <div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-12jlilq-MuiPaper-root"
               style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
          >
            <div class="MuiBox-root css-hpgf8j">
              <p class="MuiTypography-root MuiTypography-body1 css-12h6qye-MuiTypography-root">
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-qc0aqs-MuiGrid-root">
    <div class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root">
      <div class="MuiBox-root css-qxh7fn">
        <div class="MuiBox-root css-bmvtgo">
          Showing data for the last time point.
          <br>
          Click the chart to choose a different time point.
        </div>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root">
      <div class="MuiBox-root css-19midj6">
        <p class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-12k2fzb-MuiTypography-root"
           data-testid="empty"
           style="--Typography-textAlign: center;"
        >
          No data to show for this time point.
        </p>
      </div>
    </div>
  </div>
</div>
`;
